lms/djangoapps/discussion_api/api.py

```diff
@@ -177,103 +185,136 @@ def get_course(request, course_key):
     }
 
 
-def get_course_topics(request, course_key):
+def get_courseware_topics(request, course_key, course, topic_ids):
     """
-    Return the course topic listing for the given course and user.
+    Returns a list of topic trees for courseware-linked topics.
 
     Parameters:
 
-    course_key: The key of the course to get topics for
-    user: The requesting user, for access control
+        request: The django request objects used for build_absolute_uri.
+        course_key: The key of the course to get discussion threads for.
+        course: The course for which topics are requested.
+        topic_ids: A list of topic IDs for which details are requested.
+            This is optional. If None then all course topics are returned.
 
     Returns:
+        A list of courseware topics and a list of existing topics among
+        topic_ids.
 
-    A course topic listing dictionary; see discussion_api.views.CourseTopicViews
-    for more detail.
     """
+    courseware_topics = []
+    existing_topic_ids = []
+
     def get_module_sort_key(module):
         """
         Get the sort key for the module (falling back to the discussion_target
         setting if absent)
         """
         return module.sort_key or module.discussion_target
-    course = _get_course(course_key, request.user)
+
+    def get_sorted_modules(category):
+        """Returns key sorted modules by category"""
+        return sorted(modules_by_category[category], key=get_module_sort_key)
+
     discussion_modules = get_accessible_discussion_modules(course, request.user)
     modules_by_category = defaultdict(list)
     for module in discussion_modules:
         modules_by_category[module.discussion_category].append(module)
 
-    def get_sorted_modules(category):
-        """Returns key sorted modules by category"""
-        return sorted(modules_by_category[category], key=get_module_sort_key)
+    for category in sorted(modules_by_category.keys()):
+        children = []
+        for module in get_sorted_modules(category):
+            if not topic_ids or module.discussion_id in topic_ids:
+                discussion_topic = DiscussionTopic(
+                    module.discussion_id,
+                    module.discussion_target,
+                    get_thread_list_url(request, course_key, [module.discussion_id]),
+                )
+                children.append(discussion_topic.__dict__)
```

Comment:
Do we have a better way to serialize discussion topics?  Like maybe a DRF Serializer subclass?  This seems like it could introduce unexpected data (internal instance variables, for example).